### ISSUE TITLE
Stripe .Net SDK does not support new "collection_options" property for account link generation

### ISSUE THREAD

**Issue by eswann** [stripe/stripe-dotnet#2838]
### Describe the bug

We received an email from Stripe, stating that in order for our integration to continue functioning as is (not sure how dire that is), we would need to provide a new property to the AccountLinkCreateOptions sent in the "onboarding/account edit" link generation call. We were given two links to documentation regarding this:
https://stripe.com/docs/api/account_links/object
https://stripe.com/docs/connect/custom/hosted-onboarding#how-to-use-connect-onboarding-for-custom-accounts

It shows an example using the .Net SDK. However, the .Net SDK does not support this new "CollectionOptions" property even though the documentation states that it should. This is directly from the docs:
```
var options = new AccountLinkCreateOptions
{
    Account = "{{CONNECTED_ACCOUNT_ID}}",
    RefreshUrl = "https://example.com/reauth",
    ReturnUrl = "https://example.com/return",
    Type = "account_onboarding",
    CollectionOptions = new AccountLinkCollectionOptionsOptions
    {
        Fields = "eventually_due",
    },
};
```
 

### To Reproduce

Get latest .Net SDK and try to add "CollectionOptions" property to the AccountLinkCreationOptions.

### Expected behavior

Property should exist.

### OS

Any

### .NET version

Any

### Library version

43.11.0

### API version

2022-11-15

**Comment by remi-stripe** [comment]
@eswann The feature is brand new in our API (just shipped on Monday) and we're working on adding it to our SDKs. Hopefully we'll have a version out by Friday that supports it. Sorry about the confusion here!

**Comment by eswann** [comment]
@remi-stripe Appreciate the update. We got a pretty dire looking warning from Stripe stating that our onboarding was going to stop working as it currently does if we didn't add in this property (with links to inaccurate .Net SDK documentation). Is this just a lack of coordination or is this not as serious as they are making it sound to us? If a new Stripe feature is going to break existing functionality, it seems like it would be good to have it implemented in the SDK's before rolling it out and sending scary messages to customers. Sorry, I'm somewhat confused by this whole thing.

**Comment by remi-stripe** [comment]
@eswann unfortunately just too many things happening at once and we didn't time it right. Sorry about the hassle but we are looking into fixing this soon and improving it overall.

**Comment by eswann** [comment]
@remi-stripe You and me both.  :)

**Comment by chernovol** [comment]
@eswann @remi-stripe 
you can implement it like this
var options = new AccountLinkCreateOptions
{
    Account = "{{CONNECTED_ACCOUNT_ID}}",
    RefreshUrl = "https://example.com/reauth",
    ReturnUrl = "https://example.com/return",
    Type = "account_onboarding",
    ExtraParams = new Dictionary<string, object>()
    {
        { "collection_options[fields]", "eventually_due" },
        { "collection_options[future_requirements]", "include" }
    }
};
var service = new AccountLinkService();
var accountLink = await service.CreateAsync(options);

**Comment by eswann** [comment]
@chernovol Appreciate it, that's what I did.

**Comment by remi-stripe** [comment]
We just released the version [43.12.0](https://github.com/stripe/stripe-dotnet/blob/master/CHANGELOG.md#43120---2024-01-25) today which solves this problem! Thanks for flagging and sorry again for the delay on that one.

**Comment by eswann** [comment]
Thanks for the quick release.
